src/librustc_mir/transform/check_unsafety.rs

```diff
@@ -679,7 +678,7 @@ pub fn check_unsafety<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) {
     }
 
     let mut unsafe_blocks: Vec<_> = unsafe_blocks.into_iter().collect();
-    unsafe_blocks.sort();
+    unsafe_blocks.sort_by_cached_key(|(hir_id, _)| tcx.hir().hir_to_node_id(*hir_id));
```

Comment:
I think it's find to just use `sort` here. Do you know which errors changed order with that?

Reply:
Nope, plain sort breaks 2 issue test cases related to unused unsafe.

Reply:
Well we can land `sort_by_cached_key` here and you could open a new PR with an additional commit which changes it back to `sort`, so we can look at the failures.